### PR TITLE
Secure cookie

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

https://dimagi-dev.atlassian.net/browse/SAAS-13823

This recently added cookie did not include `secure=settings.SECURE_COOKIES`. We probably need a better process to ensure cookies are added with the correct secure value, but just pushing the fix for now.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by dannyroberts — APPROVED** [review]

**Review by esoergel — APPROVED** [review]
> We probably need a better process to ensure cookies are added with the correct secure value.

Should it _always_ be set?  Looking at [how `set_cookie` works](https://github.com/django/django/blob/main/django/http/response.py#L265-L266) I bet it'd be pretty straightforward to add in some middleware that just sets that flag to `True` for all cookies included in the response.  It'd be nicer if we could somehow subclass or patch `HttpResponse` versus needing yet another middleware layer.

**Comment by gherceg** [comment]
> Should it always be set?

Yeah taken from the original ticket that flagged this issue:

> We intend to use HTTPS exclusively, so there is no reason ever not to mark a cookie as Secure.

So I agree a straightforward solution could be middleware. I haven't had to dig much into our middleware so I might have some naive optimism, but I actually like the sound of middleware better than subclassing/patching `HttpResponse`. I'll merge this for now and continue that discussion on the ticket itself.

File: corehq/apps/settings/views.py [hunk 1]
```diff
@@ -252,7 +252,8 @@ def post(self, request, *args, **kwargs):
             self.settings_form.update_user()
 
             res = redirect(reverse(MyAccountSettingsView.urlname))
-            res.set_cookie(settings.LANGUAGE_COOKIE_NAME, self.request.couch_user.language)
+            res.set_cookie(settings.LANGUAGE_COOKIE_NAME, self.request.couch_user.language,
+                           secure=settings.SECURE_COOKIE)
```
Comment (snopoke) [inline corehq/apps/settings/views.py:256]: @gherceg this should be `SECURE_COOKIES`: https://sentry.io/organizations/dimagi/issues/3494689565/

Comment (gherceg) [reply corehq/apps/settings/views.py:256]: 🤦🏻 thanks for catching this @snopoke. Resolved in the bbc393a5d5c3eb6f234bf6744521a4b79e599b7f

**Pull request merged**